main/src/cgeo/geocaching/connector/lc/LCApi.java

```diff
@@ -0,0 +1,279 @@
+            cache.setDifficulty((float) 1);
+            cache.setTerrain((float) 1);
+            cache.setSize(CacheSize.getById("virtual"));
+            cache.setFound(false);
+            DataStore.saveCache(cache, EnumSet.of(SaveFlag.CACHE));
+            return cache;
+        } catch (final NullPointerException e) {
+            Log.e("LCApi.parseCache", e);
+            return null;
+        }
+    }
+
+    // Having a separate parser for details is required because the API provider
+    // decided to use different upper/lower case wordings for the same entities
+
+    @Nullable
+    private static Geocache parseCacheDetail(final JsonNode response) {
+        try {
+            final Geocache cache = new Geocache();
+            final JsonNode location = response.at("/Location");
+            final String firebaseDynamicLink = response.get("FirebaseDynamicLink").asText();
+            final String[] segments = firebaseDynamicLink.split("/");
+            final String id = segments[segments.length - 1];
+            final String uuid = response.get("Id").asText();
+            final String ilink = response.get("KeyImageUrl").asText();
+            final String desc = response.get("Description").asText();
+            cache.setReliableLatLon(true);
+            cache.setGeocode("LC" + uuid);
+            cache.setCacheId(id);
+            cache.setName(response.get("Title").asText());
+            if (Settings.useAdvancedALC()) {
+                cache.setDescription("<img src=\"" + ilink + "\" </img><p><p>" + desc);
+            } else {
+                cache.setDescription("");
+            }
+            cache.setCoords(new Geopoint(location.get("Latitude").asText(), location.get("Longitude").asText()));
+            cache.setType(CacheType.ADVLAB);
+            cache.setDifficulty((float) 1);
+            cache.setTerrain((float) 1);
+            cache.setSize(CacheSize.getById("virtual"));
+            cache.setFound(false);
+            cache.setDisabled(false);
+            cache.setHidden(parseDate(response.get("PublishedUtc").asText()));
+            cache.setOwnerDisplayName(response.get("OwnerUsername").asText());
+            cache.setWaypoints(parseWaypoints((ArrayNode) response.path("GeocacheSummaries")), false);
+            cache.setDetailedUpdatedNow();
+            DataStore.saveCache(cache, EnumSet.of(SaveFlag.DB));
+            return cache;
+        } catch (final NullPointerException e) {
+            Log.e("LCApi.parseCache", e);
+            return null;
+        }
+    }
+
+    @Nullable
+    private static List<Waypoint> parseWaypoints(final ArrayNode wptsJson) {
+        if (!Settings.useAdvancedALC()) {
+            return null;
+        }
+        List<Waypoint> result = null;
+        final Geopoint pointZero = new Geopoint(0, 0);
+        int stageCounter = 0;
+        for (final JsonNode wptResponse: wptsJson) {
+            //Log.e(wptResponse.toPrettyString());
```

Comment:
Please remove debugging code

Reply:
Done ... for the sole purpose of being in line with your guidelines. In my own projects I leave those in as I frequently need the  in the next debugging cyle ;)

Reply:
@baiti you may use `Log.d` or `Log.v` for debug/verbose logs. Then there would not be the need to comment them out and you can turn them on as needed during development or Bugfix.

Reply:
Good to know. Next time then. In order to have some filtering, would I also specify a "TAG"? I remember when I did that Kotlin thing, the  function was `Log.x(TAG, msg);` I assume I could use tag for filtering?

Reply:
@baiti cgeo has it's own Log class which does the TAG assignment internally (basically the tag is cgeo). In its methods you can't specify a tag yourself. But of course You can use certain keywords in the log message itself to search for afterwards. I often e.g. put the method name on the log message for this purpose